src/pathfinding_igvc/test/astar-test.cpp

```diff
@@ -0,0 +1,65 @@
+//
+// Created by min on 20/05/18.
+//
+
+#include <AStar.h>
+#include "PathFinderTestUtils.h"
+#include <gtest/gtest.h>
+
+TEST(AStar, FullPathTest) {
+    /* origin of OccupancyGrid */
+    // initialize origin of occupancy grid
+    geometry_msgs::Pose origin =
+            PathFinderTestUtils::constructPose(0.0, 0.0, 0.0);
+
+    /* mapMetaData of OccupancyGrid */
+    // initialize mapMetaData
+    nav_msgs::MapMetaData mapMetaData;
+    mapMetaData.resolution = 1.0;
+    mapMetaData.width = 10;
+    mapMetaData.height = 9;
+    // add origin to mapMetaData
+    mapMetaData.origin = origin;
+
+    /* OccupancyGrid */
+    // initialize occupancy grid
+    nav_msgs::OccupancyGrid grid;
+
+    // set mapMetaData
+    grid.info = mapMetaData;
+    grid.data = {
+            AStar::GRID_FREE, AStar::GRID_FREE, AStar::GRID_FREE, AStar::GRID_OCCUPIED, AStar::GRID_OCCUPIED, AStar::GRID_OCCUPIED, AStar::GRID_FREE, AStar::GRID_OCCUPIED, AStar::GRID_OCCUPIED, AStar::GRID_FREE,
```

Comment:
I really like this test, I think adding the following to the start of the file might make this diagram easier to read

```
typedef AStart::GRID_FREE _;
typedef AStar::GRID_OCCUPIED X;
```
And then changing the diagram accordingly

Reply:
wow good idea